src/api/vdom.js

```diff
@@ -166,17 +166,25 @@ function wrapIdomFunc(func, tnameFuncHandler = () => {}) {
       const isElementClosing = func === elementClose;
       const isElementOpening = func === elementOpen;
 
-      // If we're skipping the tree, we must skip everything except for the
-      // closing of the element that originally started the skipping.
-      if (skipCurrentTree && !isElementClosing && !currentElement()[$skipCurrentElement]) {
-        return;
+      if (isElementOpening) {
+        if (skipCurrentTree) {
+          return;
+        } else {
+          const elem = func(...args);
+          skipCurrentTree = !!elem[$skipCurrentElement];
+          return elem;
+        }
       }
+      
+      if (isElementClosing) {
+        let elem = currentElement();
 
-      const elem = func(...args);
+        if (elem[$skipCurrentElement]) {
+          skipCurrentTree = false;
+        } else {
```

Comment:
K @bradleyayers updated the PR. Logic changed substantially because we now have to keep track of how many levels deep past the skipped element because `currentElement()` doesn't actually return the element we think it does. In IDOM master `currentPointer()` would be used.